pkg/network/protocols/http2/protocol.go

```diff
@@ -254,7 +287,48 @@ func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
 		return started > 0 && (now-started) > ttl
 	})
 
-	p.mapCleaner = mapCleaner
+	p.http2InFlightMapCleaner = mapCleaner
+}
+
+func (p *protocol) setupDynamicTableMapCleaner(mgr *manager.Manager) {
+	dynamicTableMap, _, err := mgr.GetMap(dynamicTable)
+	if err != nil {
+		log.Errorf("error getting %q map: %s", dynamicTable, err)
+		return
+	}
+
+	mapCleaner, err := ddebpf.NewMapCleaner(dynamicTableMap, new(http2DynamicTableIndex), new(http2DynamicTableEntry))
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return
+	}
+
+	var terminatedConnections []netebpf.ConnTuple
+	terminatedConnectionsMap := make(map[netebpf.ConnTuple]struct{})
+	mapCleaner.Clean(p.cfg.HTTP2DynamicTableMapCleanerInterval,
+		func() bool {
+			p.terminatedConnectionsEventsConsumer.Sync()
+			terminatedConnections = terminatedConnections[:0]
+			p.terminatedConnectionMux.Lock()
+			terminatedConnections = append(terminatedConnections, p.terminatedConnections...)
```

Comment:
Do you need the intermediate slice, or could you populate the `terminatedConnectionsMap` here?